integration-cli/docker_cli_build_test.go

```diff
@@ -7357,3 +7357,19 @@ func (s *DockerSuite) TestBuildWindowsEnvCaseInsensitive(c *check.C) {
 		c.Fatalf("Case insensitive environment variables on Windows failed. Got %s", res)
 	}
 }
+
+func (s *DockerSuite) TestBuildLabelsArgs(c *check.C) {
```

Comment:
I agree this should really be a unit test.  Every bug is technically invoked from either the CLI or API, but I don't think that necessarily warrants an integration test. Integration tests are great for catching regressions where the bug was because of an integration between two components.

In this case the bug seems to be in a single component, so it could be tested directly.

A rule I like to follow is that if a function is hard to unit test, it's a good sign the function should be refactored in a way that makes it easier to test. Functions that are hard to test are often hard to re-use as well.  Since this function is 100 lines long, it is difficult to test the behaviour in isolation. It might not be a bad idea to split it up, making it easier to test.